### ISSUE TITLE
pip install dynet not working on Mac

### ISSUE THREAD

**Issue by chunyang-wen** [clab/dynet#1034]
Mac OS:10.12.5 (16F73)

```python
pip install dynet
```

```python
>>> import dynet
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/dynet.py", line 15, in <module>
    from _dynet import *
ImportError: dlopen(/usr/local/lib/python2.7/site-packages/_dynet.so, 2): Library not loaded: /private/var/folders/xf/qm8tkgkx5yjfgrhrmqxvsknr0000gn/T/pip-build-snn3Sx/dynet/build/py2.7-64bit/dynet/libdynet.dylib
  Referenced from: /usr/local/lib/python2.7/site-packages/_dynet.so
  Reason: image not found
```

**Comment by neubig** [comment]
I was able to confirm this problem, @danielhers any ideas?

**Comment by danielhers** [comment]
It's installing from source - since I wasn't able to build macOS wheels with the library in them yet, I didn't upload them. A workaround would be something like

  pip install --build-dir=~/dynet_build dynet
  export DYLD_LIBRARY_PATH=~/dynet_build/dynet/build/dynet/:$DYLD_LIBRARY_PATH

**Comment by danielhers** [comment]
Please try again now. I think `2.0.1.post9` wheels should include `libdynet.dylib`.

**Comment by chunyang-wen** [comment]
Hmm. It still does not work even through the file exists.

**Comment by danielhers** [comment]
Right. Looks like the `rpath` wasn't set to find it though. Added `@loader_path/` to the `rpath` in  `2.0.1.post10` but I'm still not sure it worked.

**Comment by chunyang-wen** [comment]
```shell
export DYLD_LIBRARY_PATH=/install/path/of/dynet/:$DYLD_LIBRARY_PATH
```
will solve the problem

**Comment by danielhers** [comment]
Fixed in `2.0.1.post13` and verified. `DYLD_LIBRARY_PATH` is no longer required.
